gammapy/time/lightcurve.py

```diff
@@ -320,6 +321,124 @@ def create_fixed_time_bin(time_step, spectrum_extraction):
+                                            spectrum_extraction, separators=[]):
+        """Create time intervals such that each bin of a light curve reach a given significance.
+
+        Parameters
+        ----------
+        significance : float
+            Target significance for each light curve point
+        significance_method:
+            Select the method used to compute the significance ### unused yet ###
+        energy_range : `~astropy.units.Quantity`
+            True energy range to evaluate integrated flux (true energy)
+        spectrum_extraction : `~gammapy.spectrum.SpectrumExtraction`
+            Contains statistics, IRF and event lists
+        separators : `list` of float
+            Contains a list of time (in MJD) to break the current point creation
+            
+        Returns
+        -------
+        intervals : `list` of `~astropy.time.Time`
+            List of time intervals
+        """
+
+        def gettime(item):
+            return item[0]
+
+        def in_list(item, L):
+            o, j = np.where(L == item)
+            for index in o:
+                yield index
+
+        time_holder = []
+        obs_properties = []
+        n_obs = 0
+
+        for time in separators:
+            time_holder.append([time, 'break'])
+
+        for obs in spectrum_extraction.obs_list:
+            # start and end will need to be redefined once the data storage format is fixed
+            time_holder.append([obs.events.time.min().value - 0.0000001, 'start'])
+            time_holder.append([obs.events.time.max().value + 0.0000001, 'end'])
+            obs_properties.append([obs.observation_dead_time_fraction, spectrum_extraction.bkg_estimate[n_obs].a_off])
+            n_obs += 1
+
+        # prepare the on and off photon list as in the flux point computation -> should be updated accordingly
+        for t_index, obs in enumerate(self.obs_list):
+            spec = self.obs_spec[t_index]
+            # get ON and OFF evt list
+            off_evt = self.off_evt_list[t_index]
+            on_evt = self.on_evt_list[t_index]
+            # prepare energy range filter
+            e_reco = spec.e_reco
+            emin = e_reco[e_reco.searchsorted(max(spec.lo_threshold, energy_range[0]))]
+            emax = e_reco[e_reco.searchsorted(min(spec.hi_threshold, energy_range[1])) - 1]
+            # filter ON events
+            on = on_evt.select_energy([emin, emax])
+            on = on.select_energy(energy_range)
+            # filter OFF events
+            off = off_evt.select_energy([emin, emax])
+            off = off.select_energy(energy_range)
+
+            for time in on.time:
+                time_holder.append([time.value, 'on'])
+            for time in off.time:
+                time_holder.append([time.value, 'off'])
+
+        # sort all elements in the table by time
+        time_holder = sorted(time_holder, key=gettime)
```

Comment:
Suggest to just define this simple key function inline like this (and remove the `def gettime` above):
```
time_holder = sorted(time_holder, key=lambda item: item[0])
```